tephra-core/src/main/java/co/cask/tephra/TransactionManager.java

```diff
@@ -905,6 +906,11 @@ private void doCommit(long writePointer, Set<ChangeId> changes, long commitPoint
     }
     // moving read pointer
     moveReadPointerIfNeeded(writePointer);
+    if (checkpointWritePointers != null) {
+      for (long checkpointWritePointer : checkpointWritePointers) {
```

Comment:
Yes - you are right, we can just advance using the write pointer. After making changes for checkpointing, I think we missed going over the code and reconciling write pointers and transaction ids.